test/unittests/stack_test.cpp

```diff
@@ -71,62 +68,99 @@ TEST(stack, clear)
     EXPECT_EQ(stack.size(), 0);
 }
 
-TEST(stack, resize)
+TEST(stack, clear_on_empty)
 {
     Stack<char> stack;
-    stack.push('a');
-    stack.push('b');
-
-    EXPECT_FALSE(stack.empty());
-    EXPECT_EQ(stack.size(), 2);
-
-    // grow stack
-    stack.resize(4);
-    EXPECT_FALSE(stack.empty());
-    EXPECT_EQ(stack.peek(), 0);
-    EXPECT_EQ(stack.peek(1), 0);
-    EXPECT_EQ(stack.peek(2), 'b');
-    EXPECT_EQ(stack.peek(3), 'a');
-    EXPECT_EQ(stack.size(), 4);
-
-    stack.drop();
-    EXPECT_FALSE(stack.empty());
-    EXPECT_EQ(stack.size(), 3);
-
-    // shrink stack
-    stack.resize(1);
-    EXPECT_FALSE(stack.empty());
-    EXPECT_EQ(stack.size(), 1);
-
     stack.clear();
-    EXPECT_TRUE(stack.empty());
 }
 
-TEST(stack, iterator)
+TEST(operand_stack, construct)
 {
-    Stack<char> stack;
-    stack.push('a');
-    stack.push('b');
-    stack.push('c');
+    OperandStack stack{0};
```

Comment:
Nit: as with other containers I'd prefer parenthesis here not to confuse with initializer list.
In execute it looks ok because it uses `max_stack_height`, but here it looks like stack containing 0.
```suggestion
    OperandStack stack(0);
```

Reply:
Done.